xcell/mappers/mapper_HSC_DR1wl.py

```diff
@@ -0,0 +1,293 @@
+                cat['e1'] = e1
+                cat['e2'] = e2
+
+                # Remove unnecessary columns
+                cat.keep_columns(['ra', 'dec', 'e1', 'e2', self.w_name])
+                cats.append(cat)
+            self.cat = vstack(cats)
+            if fname_lite is not None:
+                self.cat.write(fname_lite)
+        return self.cat
+
+    def _clean_raw_catalog(self, fnames):
+        cats = []
+        for fname in fnames:
+            if not os.path.isfile(fname):
+                raise ValueError(f"File {fname} not found")
+            c = Table.read(fname)
+            sel = np.ones(len(c), dtype=bool)
+            isnull_names = []
+            for key in c.keys():
+                if key.__contains__('isnull'):
+                    if not key.startswith('ishape'):
+                        sel[c[key]] = 0
+                    isnull_names.append(key)
+                else:
+                    # Keep photo-zs and shapes even if they're NaNs
+                    if ((not key.startswith("pz_")) and
+                            (not key.startswith('ishape'))):
+                        sel[np.isnan(c[key])] = 0
+            c.remove_columns(isnull_names)
+            c.remove_rows(~sel)
+
+            # Collect sample cuts
+            sel_area = c['wl_fulldepth_fullcolor']
+            sel_clean = sel_area & c['clean_photometry']
+            sel_maglim = np.ones(len(c), dtype=bool)
+            sel_maglim[c['icmodel_mag'] -
+                       c['a_i'] > self.icut] = 0
+            # Blending
+            sel_blended = np.ones(len(c), dtype=bool)
+            # Shear sample cuts as defined in https://arxiv.org/abs/1705.06745
+            # abs_flux<10^-0.375
+            sel_blended[c['iblendedness_abs_flux'] >= 0.42169650342] = 0
+            # S/N in i
+            sel_fluxcut_i = np.ones(len(c), dtype=bool)
+            sel_fluxcut_i[c['icmodel_flux'] < 10*c['icmodel_flux_err']] = 0
+            # S/N in g
+            sel_fluxcut_g = np.ones(len(c), dtype=int)
+            sel_fluxcut_g[c['gcmodel_flux'] < 5*c['gcmodel_flux_err']] = 0
+            # S/N in r
+            sel_fluxcut_r = np.ones(len(c), dtype=int)
+            sel_fluxcut_r[c['rcmodel_flux'] < 5*c['rcmodel_flux_err']] = 0
+            # S/N in z
+            sel_fluxcut_z = np.ones(len(c), dtype=int)
+            sel_fluxcut_z[c['zcmodel_flux'] < 5*c['zcmodel_flux_err']] = 0
+            # S/N in y
+            sel_fluxcut_y = np.ones(len(c), dtype=int)
+            sel_fluxcut_y[c['ycmodel_flux'] < 5*c['ycmodel_flux_err']] = 0
+            # S/N in grzy (at least 2 pass)
+            sel_fluxcut_grzy = (sel_fluxcut_g+sel_fluxcut_r +
+                                sel_fluxcut_z+sel_fluxcut_y >= 2)
+            # Overall S/N
+            sel_fluxcut = sel_fluxcut_i*sel_fluxcut_grzy
+            # Stars
+            sel_stars = np.ones(len(c), dtype=bool)
+            sel_stars[c['iclassification_extendedness'] > 0.99] = 0
```

Comment:
I think stars are `iclassification_extendedness = 0` (last line of p.16 in the paper you cite above). Why do you use 0.99?

Reply:
right now this is a binary value (0 or 1), so it doesn't really matter, but in future data releases this will be a continuous variable and one needs to cut at some point. 0.99 is just a reminder of that (and right now it doesn't matter)